pkg/util/kube/predicate.go

```diff
@@ -98,10 +98,44 @@ func NewGenericEventPredicate(f func(object client.Object) bool) predicate.Predi
 }
 
 // NewUpdateEventPredicate creates a new Predicate that checks the Update event with the provided func
-func NewUpdateEventPredicate(f func(objectOld client.Object, objectNew client.Object) bool) predicate.Predicate {
+func NewUpdateEventPredicate(
+	f func(objectOld client.Object, objectNew client.Object) bool,
+) predicate.Predicate {
 	return predicate.Funcs{
 		UpdateFunc: func(event event.UpdateEvent) bool {
 			return f(event.ObjectOld, event.ObjectNew)
 		},
+		CreateFunc: func(event event.CreateEvent) bool {
```

Comment:
We should only care about the `Update()` for `UpdateEventPredicate`, similar for `DeleteEventPredicate`